src/test/java/com/fasterxml/jackson/databind/introspect/Transient3948Test.java

```diff
@@ -0,0 +1,92 @@
+package com.fasterxml.jackson.databind.introspect;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.BaseMapTest;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.Serial;
+import java.io.Serializable;
+
+public class Transient3948Test extends BaseMapTest {
+
+    public static class Obj implements Serializable {
```

Comment:
Need to add `@JsonPropertyOrder` of some kind to ensure specific serialization order.